clang/lib/Driver/Driver.cpp

```diff
@@ -4340,6 +4361,17 @@ class OffloadingActionBuilder final {
       }
     }
 
+    void addDeviceLinkDependences(OffloadDepsJobAction *DA) override {
+      for (unsigned I = 0; I < ToolChains.size(); ++I) {
+        // Register dependent toolchain.
+        DA->registerDependentActionInfo(
+            ToolChains[I], /*BoundArch=*/StringRef(), Action::OFK_SYCL);
+
```

Comment:
add `assert(ToolChains[I]->getTriple().isSPIR())`?

Reply:
Hm, I think this would not be correct. As I understand for non-SPIRV targets (f.e cuda), linking is also done on bitcode inputs and backend is then invoked after linking.

Reply:
Oh - yes, that's right.